tests/Oryx.Integration.Tests/LocalDockerTests/NodeEndToEndTests.cs

```diff
@@ -91,6 +91,63 @@ public async Task CanBuildAndRunNodeApp_UsingZippedNodeModules()
                 });
         }
 
+        [Fact]
+        public async Task CanBuildAndRunNodeApp_UsingZippedNodeModules_WithoutExtracting()
+        {
+            // NOTE:
+            // 1. Use intermediate directory(which here is local to container) to avoid errors like
+            //      "tar: node_modules/form-data: file changed as we read it"
+            //    related to zipping files on a folder which is volume mounted.
+            // 2. Use output directory within the container due to 'rsync' having issues with volume mounted directories
+
+            // Arrange
+            var appOutputDirPath = Directory.CreateDirectory(Path.Combine(_tempRootDir, Guid.NewGuid().ToString("N"))).FullName;
+            var appOutputDirVolume = DockerVolume.Create(appOutputDirPath);
+            var appOutputDir = appOutputDirVolume.ContainerDir;
+            var nodeVersion = "10.14";
+            var appName = "webfrontend";
+            var hostDir = Path.Combine(_hostSamplesDir, "nodejs", appName);
+            var volume = DockerVolume.Create(hostDir);
+            var appDir = volume.ContainerDir;
+            var containerPort = "80";
+            var portMapping = $"{HostPort}:{containerPort}";
+            var runAppScript = new ShellScriptBuilder()
+                .AddCommand("export ORYX_DISABLE_NODE_MODULES_EXTRACT=true")
+                .AddCommand($"cd {appDir}")
+                .AddCommand($"oryx -appPath {appOutputDir} -bindPort {containerPort}")
+                .AddCommand(DefaultStartupFilePath)
+                .AddDirectoryDoesNotExistCheck("/node_modules")
```

Comment:
move this condition one line above

Reply:
I can't, since I want to make sure the startup script itself doesn't create the node_modules folder from the tar.gz file

Reply:
Ah, my bad...incorrectly thought the oryx command was even executing!